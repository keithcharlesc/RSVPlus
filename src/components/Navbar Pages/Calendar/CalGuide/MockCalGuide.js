import React, { useState } from "react";
import MockNavigationBar from "../../NavigationBar/MockNavigationBar";
import { Carousel } from "react-bootstrap";
import StepOne from "./img/StepOne.png";
import StepTwo from "./img/StepTwo.png";
import StepThree from "./img/StepThree.png";
import { BrowserRouter, Link } from "react-router-dom";
import "./CalGuide.css";

export default function CalGuide() {
  const [index, setIndex] = useState(0);

  const handleSelect = (selectedIndex, e) => {
    setIndex(selectedIndex);
  };

  return (
    <>
      <MockNavigationBar />
      <div
        className="p-3 mb-2 bg-dark text-white"
        style={{ minHeight: "100vh" }}
      >
        <h2 className="text-center mb-5">Calendar How-To</h2>
        <BrowserRouter>
          <Link to="/calendar">Back to Calendar</Link>
        </BrowserRouter>
        <Carousel activeIndex={index} onSelect={handleSelect}>
          <Carousel.Item>
            <img className="d-block w-100" src={StepOne} alt="First slide" />
            <Carousel.Caption>
              <h3 className="mt-5">Step 1</h3>
              <p className="mt-5">Going to Google Calendar.</p>
            </Carousel.Caption>
          </Carousel.Item>
          <Carousel.Item>
            <img className="d-block w-100" src={StepTwo} alt="Second slide" />

            <Carousel.Caption>
              <h3 className="mt-5">Step 2</h3>
              <p className="mt-5">Getting public URL.</p>
            </Carousel.Caption>
          </Carousel.Item>
          <Carousel.Item>
            <img className="d-block w-100" src={StepThree} alt="Third slide" />

            <Carousel.Caption>
              <h3 className="mt-5">Step 3</h3>
              <p className="mt-5">
                Saving URL in RSVP+ for future logins to fetch from.
              </p>
            </Carousel.Caption>
          </Carousel.Item>
        </Carousel>
      </div>
    </>
  );
}
