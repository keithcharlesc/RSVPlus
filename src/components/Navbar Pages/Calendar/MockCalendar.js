import React, { useState } from "react";
import { Form, Button, Container, Row, Col, Alert } from "react-bootstrap";
import MockNavigationBar from "../NavigationBar/MockNavigationBar";
import "./Calendar.css";
import GoogleCal from "./GoogleCal";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

/*----- GAPI------*/
var gapi = { auth2: {}, client: {} };
var CLIENT_ID =
  "1011248109211-umpu5g48dj5p4hqlnhvuvl6f4c9qdhn3.apps.googleusercontent.com";
var API_KEY = "AIzaSyD3-pnAPnBMzBqL_dAZYYyVGretc42zUnA";
var DISCOVERY_DOCS = [
  "https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest",
];
var SCOPES = "https://www.googleapis.com/auth/calendar.events";

gapi.load = (a, f) => f();
/*---------------*/

export default function Calendar() {
  const [startDate, setStartDate] = useState(new Date());
  const [endDate, setEndDate] = useState(new Date());
  const [loading, setLoader] = useState(false);
  const [error, setError] = useState("");
  const [success, setSuccess] = useState("");

  /* ---------------- ADDING EVENT -------------*/
  const handleClick = (e) => {
    e.preventDefault();
    setLoader(true);
    setError("");
    setSuccess("");

    gapi.auth2
      .getAuthInstance()
      .then(() => {
        var event = {
          summary: "data.name",
          location: "data.location",
          description: "data.description",
          start: {
            dateTime:
              startDate.toLocaleDateString("en-CA") +
              "T" +
              startDate.toLocaleTimeString("it-IT").slice(0, 5) +
              ":00+08:00", //requires time input
            timeZone: "Asia/Singapore",
          },
          end: {
            dateTime:
              endDate.toLocaleDateString("en-CA") +
              "T" +
              endDate.toLocaleTimeString("it-IT").slice(0, 5) +
              ":00+08:00", //requires time input
            timeZone: "Asia/Singapore",
          },
          reminders: {
            useDefault: false,
            overrides: [
              { method: "email", minutes: 24 * 60 },
              { method: "popup", minutes: 10 },
            ],
          },
        };

        if (
          startDate.toLocaleDateString("en-CA") ===
          endDate.toLocaleDateString("en-CA")
        ) {
          let startDateHoursAndMinutes = startDate
            .toLocaleTimeString("it-IT")
            .slice(0, 5);

          let endDateHoursAndMinutes = endDate
            .toLocaleTimeString("it-IT")
            .slice(0, 5);

          let minutesForStartDate = getMinutes(startDateHoursAndMinutes);
          let minutesForEndDate = getMinutes(endDateHoursAndMinutes);

          if (minutesForEndDate < minutesForStartDate) {
            setError(
              "Error! Start Time must be earlier than End Time for same date!"
            );
            return;
          }
        }

        var request = gapi.client.calendar.events.insert({
          calendarId: "primary",
          resource: event,
        });

        setSuccess("Event successfully created!");

        request.execute((event) => {
          //console.log(event);
          window.open(event.htmlLink);
        });
      })
      .then(() => {
        setLoader(false);
      });
  };
  /* ----------Handle Form Input to use in API---------*/
  const [data, setData] = useState({
    name: "",
    description: "",
    location: "",
  });

  function handleSubmitForm(e) {
    const newData = { ...data };
    newData[e.target.id] = e.target.value;
    setData(newData);
    //console.log(newData);
  }
  /*---------------------------------------------------*/
  //Get minutes from HH:MM (hours and minutes 24 hr form)
  function getMinutes(time) {
    let str = time;
    let arr = str.split(":");
    let minutes = +arr[0] * 60 + +arr[1];
    //console.log(minutes);
    return minutes;
  }
  /*---------------------------------------------------*/
  return (
    <>
      <MockNavigationBar />
      <div
        className="p-3 mb-2 bg-dark text-white"
        style={{ minHeight: "100vh" }}
      >
        <h2 className="page-header text-center mb-5">CALENDAR</h2>

        <Container className="" style={{ minHeight: "75vh" }}>
          <Row>
            <Col md={7} className="mb-3">
              <GoogleCal />
            </Col>
            <Col md={{ span: 4, offset: 1 }}>
              <section className="create-event-section">
                <Form className="create-event-form">
                  <h1 className="mb-1"> Create an Event </h1>
                  <h6 className="text-center mb-3">(Google Calendar)</h6>
                  {success && (
                    <Alert
                      className="d-flex align-items-center justify-content-center"
                      variant="success"
                    >
                      {success}
                    </Alert>
                  )}
                  {error && <Alert variant="danger">{error}</Alert>}
                  <div className="input-group">
                    <label>Name</label>
                    <input
                      onChange={(e) => handleSubmitForm(e)}
                      id="name"
                      value={data.name}
                      placeholder=""
                      type="text"
                    ></input>
                  </div>
                  <div className="input-group">
                    <label>Description</label>
                    <input
                      onChange={(e) => handleSubmitForm(e)}
                      id="description"
                      value={data.description}
                      placeholder=""
                      type="text"
                    ></input>
                  </div>
                  <div className="input-group">
                    <label>Location</label>
                    <input
                      onChange={(e) => handleSubmitForm(e)}
                      id="location"
                      value={data.location}
                      placeholder=""
                      type="text"
                    ></input>
                  </div>
                  <div className="input-group">
                    <label>Start Date</label>
                    <DatePicker
                      selected={startDate}
                      minDate={new Date()}
                      required
                      placeholderText=""
                      onChange={(date) => setStartDate(date)}
                      timeInputLabel="Time:"
                      dateFormat="MM/dd/yyyy h:mm aa"
                      showTimeInput
                    />
                  </div>
                  <div className="input-group">
                    <label>End Date</label>
                    <DatePicker
                      selected={endDate}
                      minDate={new Date()}
                      required
                      onChange={(date) => setEndDate(date)}
                      timeInputLabel="Time:"
                      dateFormat="MM/dd/yyyy h:mm aa"
                      showTimeInput
                    />
                  </div>
                  <Button
                    data-testid="buttonToTest"
                    className="mt-2"
                    disabled={loading}
                    variant="danger"
                    style={{ width: 110, height: 50 }}
                    onClick={(e) => handleClick(e)}
                  >
                    Add Event
                  </Button>
                </Form>
              </section>
            </Col>
          </Row>
          {/* CREATE EVENT FORM */}

          {/* END OF CREATE EVENT FORM */}
        </Container>
      </div>
    </>
  );
}
