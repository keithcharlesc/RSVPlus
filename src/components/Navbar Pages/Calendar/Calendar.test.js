import React from "react";
import MockCalendar from "./MockCalendar";
import MockCalGuide from "./CalGuide/MockCalGuide";
import MockGoogleCal from "./MockGoogleCal";
import { render, getByTestId } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { toHaveTextContent } from "@testing-library/jest-dom";

it("renders CalGuide without crashing and has correct text", () => {
  const { getByText } = render(<MockCalGuide />);
  getByText("Calendar How-To");
  getByText("Back to Calendar");
  expect(getByText("Back to Calendar").href).toBe("http://localhost/calendar");
  getByText("Step 1");
  getByText("Going to Google Calendar.");
  getByText("Step 2");
  getByText("Getting public URL.");
  getByText("Step 3");
  getByText("Saving URL in RSVP+ for future logins to fetch from.");
});

it("matches snapshot", () => {
  const tree = TestRenderer.create(<MockCalGuide />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders GoogleCal without crashing and has correct text", () => {
  const { getByText } = render(<MockGoogleCal />);
  getByText("Input Google Calendar Link Below:");
  getByText("Guide");
  expect(getByText("Guide").href).toBe("http://localhost/calendar-guide");
  expect(getByTestId(document.documentElement, "buttonOne")).toHaveTextContent(
    "Reset"
  );
  expect(getByTestId(document.documentElement, "buttonTwo")).toHaveTextContent(
    "SaveURL"
  );
  expect(
    getByTestId(document.documentElement, "buttonThree")
  ).toHaveTextContent("Fetch");
  expect(getByTestId(document.documentElement, "buttonFour")).toHaveTextContent(
    "Use Sample Calendar"
  );
});

it("matches snapshot", () => {
  const tree = TestRenderer.create(<MockGoogleCal />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const { getByText } = render(<MockCalendar />);
  getByText("CALENDAR");
  getByText("Create an Event");
  getByText("(Google Calendar)");
  getByText("Name");
  getByText("Description");
  getByText("Location");
  getByText("Start Date");
  getByText("End Date");
  expect(
    getByTestId(document.documentElement, "buttonToTest")
  ).toHaveTextContent("Add Event");
});
