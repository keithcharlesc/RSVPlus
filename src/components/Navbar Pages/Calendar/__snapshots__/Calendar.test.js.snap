// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="nav-bar-appName text-danger navbar-brand"
        href="/"
      >
        RSVP+
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/channels"
            href="/channels"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Channels
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/calendar"
            href="/calendar"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Calendar
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/faq"
            href="/faq"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            FAQ
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/contact-us"
            href="/contact-us"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Contact Us
          </a>
        </div>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="nav-bar-name navbar-text"
          >
            Signed in as: 
            Bob
            <img
              className="photo ml-2 rounded-circle"
              src="thttp://email.com/image"
            />
          </span>
        </div>
      </div>
      <button
        className="ml-3 btn btn-outline-primary"
        data-testid="button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Log Out
      </button>
    </nav>
    <div>
      
    </div>
  </div>,
  <div
    className="p-3 mb-2 bg-dark text-white"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <h2
      className="text-center mb-5"
    >
      Calendar How-To
    </h2>
    <a
      href="/calendar"
      onClick={[Function]}
    >
      Back to Calendar
    </a>
    <div
      className="carousel slide"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <ol
        className="carousel-indicators"
      >
        <li
          className="active"
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
      </ol>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
        >
          <img
            alt="First slide"
            className="d-block w-100"
            src="StepOne.png"
          />
          <div
            className="carousel-caption"
          >
            <h3
              className="mt-5"
            >
              Step 1
            </h3>
            <p
              className="mt-5"
            >
              Going to Google Calendar.
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Second slide"
            className="d-block w-100"
            src="StepTwo.png"
          />
          <div
            className="carousel-caption"
          >
            <h3
              className="mt-5"
            >
              Step 2
            </h3>
            <p
              className="mt-5"
            >
              Getting public URL.
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Third slide"
            className="d-block w-100"
            src="StepThree.png"
          />
          <div
            className="carousel-caption"
          >
            <h3
              className="mt-5"
            >
              Step 3
            </h3>
            <p
              className="mt-5"
            >
              Saving URL in RSVP+ for future logins to fetch from.
            </p>
          </div>
        </div>
      </div>
      <a
        className="carousel-control-prev"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
      <a
        className="carousel-control-next"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </div>
  </div>,
]
`;

exports[`matches snapshot 2`] = `
Array [
  <form>
    <label>
      <strong>
        Input Google Calendar Link Below:
      </strong>
      <a
        className="ml-2"
        href="/calendar-guide"
        onClick={[Function]}
      >
        Guide
      </a>
      <div
        className="mt-2 row"
      >
        <div
          className="col-sm-9"
        >
          <input
            className="mb-1"
            onChange={[Function]}
            placeholder="Paste a public URL of your calendar"
            style={
              Object {
                "width": "105%",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="col-sm-3"
        >
          <button
            className="d-flex align-items-center justify-content-center ml-2 btn btn-danger"
            data-testid="buttonOne"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "height": 30,
                "width": 74,
              }
            }
            type="button"
          >
            Reset
          </button>
        </div>
      </div>
      <div
        className="mt-3 row"
      >
        <div
          className="col"
        >
          <button
            className="d-flex align-items-center justify-content-center btn btn-danger"
            data-testid="buttonTwo"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "height": 30,
              }
            }
            type="button"
          >
            SaveURL
          </button>
        </div>
        <div
          className="col"
        >
          <button
            className="d-flex align-items-center justify-content-center mb-1 btn btn-danger"
            data-testid="buttonThree"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "height": 30,
              }
            }
            type="button"
          >
            Fetch
          </button>
        </div>
        <div
          className="col"
        >
          <button
            className="d-flex align-items-center justify-content-center btn btn-danger"
            data-testid="buttonFour"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "height": 30,
                "width": 200,
              }
            }
            type="button"
          >
            Use Sample Calendar
          </button>
        </div>
      </div>
    </label>
  </form>,
  <hr />,
  <div
    style={
      Object {
        "height": "500px",
      }
    }
  >
    <p>
      Please, provide public URL of your Google calendar
    </p>
  </div>,
]
`;
